rad/prelude/cmd-parsing.rad

```diff
@@ -0,0 +1,69 @@
+{:module 'prelude/cmd-parsing
+ :doc "Functions for parsing commands."
+ :exports
+ '[parse-failure /cmd-0 /cmd-1 /cmd-2 /cmd-2-opts]}
+
+(load! "rad/prelude.rad")
+(import prelude/validation :as 'validation)
+
+(def parse-failure
+ (fn [error help]
+   (put-str! error)
+   (put-str! help)
+   (exit! 1)
+ ))
+
+ (def missing-arg-error
+   (fn [arg cmd]
+     (string-append "Missing argument \"" (show arg) "\" for command \"" cmd "\"")))
+
+ (def too-many-args-error
+   (fn [cmd]
+     (string-append "Too many arguments for command \"" cmd "\"")))
+
+(def /cmd-2-opts
+  (fn [cmd chain-name opts valid-opts opts-map help]
+    (fn [v]
+      (match v
+        [cmd] (parse-failure (string-append "Missing arguments for command " cmd) help)
+        (/cons cmd (/cons 'c-n 'os)) (if (member? c-n valid-opts)
+                                       (parse-failure (missing-arg-error chain-name cmd) help)
+                                       [:just {chain-name c-n
+                                               opts    (opts-map os)}])
+        _ :nothing))))
+
+(def /cmd-2
```

Comment:
Can you add a docstring to this and any other exported function?

Reply:
Added docstrings. Let me know if they should be more detailed.